test_api.py

```diff
@@ -31,56 +21,60 @@ def test_duplicate_headers():
 
     """
 
-    resp = request_raw_url(API_URL)
-    headers = resp.getheaders()
-    assert len(set(headers)) == len(headers), 'Duplicate response headers'
+    # Get the raw list of headers (field values not yet merged)
+    raw_headers = requests.get(API_URL).raw.headers
+
+    # Look for duplicate field values
+    for field_name, field_values in raw_headers.iteritems():
+        for value in field_values:
+            assert field_values.count(value) == 1
```

Comment:
I don't understand why you changed the perfectly working code here. This is just wrong. It checks if the same character occurs twice in a header filed value. I really don't think that's what we want to check :-(.

What this check is meant to avoid is having the exact same header line twice in the response.

If this was done because the `set(headers)` was hard to understand, we can add a comment that `set(list)` is the canonical way to make a list unique in Python. And I think checking that a list has the same length as before is the most intuitive way to assert that it already was unique. And this is what this test is about: the raw header list elements should be unique.

Reply:
> I don't understand why you changed the perfectly working code here. This is just wrong. It checks if the same character occurs twice in a header filed value. I really don't think that's what we want to check :-(.

Are you sure? `field_values` is a list, not a string. See https://github.com/urllib3/urllib3/blob/4953f29c0a0d06572708914cf02e22663e08cf55/src/urllib3/_collections.py#L397-L402